lib/improver/nbhood.py

```diff
@@ -249,8 +361,18 @@ def run(cube, radius_in_km):
             Cube containing the smoothed field after the square neighbourhood
             method has been applied.
         """
+        if isinstance(cube.data, np.ma.MaskedArray):
```

Comment:
Bad things happen. I don't think the `cumulate_array` method is ready for NaN or MDI data yet. `cumulate_array` will add up numerical mdi's as though they are valid, and will set everything below and to the right of a NaN to NaN. Then the input to `mean_over_neighbourhood` is effectively useless.

I can add a check for NaNs in the data. But amen't sure how to check for other MDI's. Can I assume that MDI's which aren't NaNs will be `np.default_fill_value` for the array?